zookeeper-server/src/main/java/org/apache/zookeeper/server/NettyServerCnxn.java

```diff
@@ -520,8 +520,12 @@ private void receiveMessage(ByteBuf message) {
                         if (len < 0 || len > BinaryInputArchive.maxBuffer) {
                             throw new IOException("Len error " + len);
                         }
-                        // checkRequestSize will throw IOException if request is rejected
-                        zkServer.checkRequestSizeWhenReceivingMessage(len);
+                        if (zkServer == null || !zkServer.isRunning()) {
+                            throw new IOException("ZK down");
+                        } else {
```

Comment:
can we move `zkServer.checkRequestSizeWhenReceivingMessage(len);` out of else block  and remove the else block.
otherwise logically there are more statements which should be in else block

Reply:
yeah, it is cleaner without the else block, I agree. And functionally equivalent. Thanks! I pushed a new commit.